### DISCUSSION TITLE
Form is flagged as dirty on reset while it's not

### DISCUSSION THREAD

**Discussion by arkadiy93** [react-hook-form/react-hook-form#12135] [Q&A]
### Version Number

7.52.1

### Codesandbox/Expo snack

[sandbox link](https://codesandbox.io/p/sandbox/react-hook-form-get-started-ts-forked-8hwpm6?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clysooj5i00063b6jchvv8jx9%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clysooj5i00023b6jg9nmyhqd%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clysooj5i00033b6j2wu718ft%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clysooj5i00053b6j26vswaf8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clysooj5i00023b6jg9nmyhqd%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clysooj5i00013b6jcimxxtf7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clysooj5i00023b6jg9nmyhqd%2522%252C%2522activeTabId%2522%253A%2522clysooj5i00013b6jcimxxtf7%2522%257D%252C%2522clysooj5i00053b6j26vswaf8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clysooj5i00043b6jo34muuhk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clysooj5i00053b6j26vswaf8%2522%252C%2522activeTabId%2522%253A%2522clysooj5i00043b6jo34muuhk%2522%257D%252C%2522clysooj5i00033b6j2wu718ft%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clysooj5i00033b6j2wu718ft%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

1. Click on reset button
2. See that the form is flagged as dirty

### Expected behaviour

I expect the form to not be dirty since state is the same.
NOTE: If I use 
`setValue("example", [], { shouldDirty: true }); `

it will work as expected and not flag it as dirty

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by arkadiy93** [comment]
@bluebill1049 this is a legit issue, why move it to discussions?

**Comment by arkadiy93** [comment]
bump

**Comment by arkadiy93** [comment]
bump

**Comment by wilav-dev** [comment]
@bluebill1049 Hello any update? I realized the issue is related with the var isEmptyResetValues because I have been using form.reset(undefined) as you suggested me some time ago, so isEmptyObject(formValues) is false
